demo/fake_server.ex

```diff
@@ -56,22 +56,97 @@ defmodule Wanda.Executions.FakeServer do
 
   defp complete_fake_execution(execution_id, group_id, targets) do
     result = Enum.random([:passing, :warning, :critical])
-    check_results = build(:check_results_from_targets, targets: targets, result: result)
+    check_results = check_results_from_targets(targets, result)
 
     build_result =
       build(:result,
         check_results: check_results,
         execution_id: execution_id,
-        group_id: group_id,
-        result: result
+        group_id: group_id
       )
 
     Executions.complete_execution!(
       execution_id,
-      build_result
+      Map.put(build_result, :result, get_result(build_result))
     )
 
     execution_completed = Messaging.Mapper.to_execution_completed(build_result)
     :ok = Messaging.publish("results", execution_completed)
   end
+
+  defp get_result(%Result{check_results: check_results}) do
+    check_results
+    |> Enum.map(& &1.result)
+    |> Enum.map(&{&1, result_weight(&1)})
+    |> Enum.max_by(fn {_, weight} -> weight end)
+    |> elem(0)
+  end
+
+  defp result_weight(:critical), do: 2
+  defp result_weight(:warning), do: 1
+  defp result_weight(:passing), do: 0
+
+  defp check_results_from_targets(targets, result) do
+    targets
+    |> Enum.flat_map(& &1.checks)
+    |> Enum.uniq()
+    |> Enum.map(fn check_id ->
+      check_targets = Enum.filter(targets, &(check_id in &1.checks))
+      check_result_from_target(check_id, check_targets, result)
+    end)
+  end
+
+  defp expectation_evaluations_from_check(check, result) do
+    Enum.map(check.expectations, fn expectation ->
+      case expectation.type do
+        :expect ->
+          build(:expectation_evaluation,
+            type: :expect,
+            name: expectation.name,
+            return_value: result == :passing
+          )
+
+        :expect_same ->
+          build(:expectation_evaluation,
+            type: :expect_same,
+            name: expectation.name,
+            return_value:
+              if result == :passing do
+                "same_value"
+              else
+                Faker.StarWars.quote()
+              end
+          )
+      end
+    end)
+  end
+
+  defp check_result_from_target(check_id, check_targets, result) do
+    {_, check} = Catalog.get_check(check_id)
```

Comment:
> In my point of view it would be better to pass the checks from top to button instead of getting them here.

I agree with this, we already load checks in the main function

Reply:
> @arbulu89 PD: I won't enter if this code should be tested or not (I think it should be)

I would test the entire module as black-box and keep those functions private, which means basically starting and execution which certain targets/checks and assert on the side-effects (message published through the mock/execution saved in the database).

Reply:
Yes, I meant that. We don't need to specifically need to test this function, but the whole faked execution code

Reply:
Got it.